service/worker/batcher/fx.go

```diff
@@ -0,0 +1,131 @@
+		dc                           *dynamicconfig.Collection
+		enabledFeature               dynamicconfig.BoolPropertyFnWithNamespaceFilter
+		enabledPerNSWorker           dynamicconfig.BoolPropertyFnWithNamespaceFilter
+		numWorkers                   dynamicconfig.IntPropertyFnWithNamespaceFilter
+		maxActivityExecutionSize     dynamicconfig.IntPropertyFnWithNamespaceFilter
+		maxWorkflowTaskExecutionSize dynamicconfig.IntPropertyFnWithNamespaceFilter
+		maxActivityPollers           dynamicconfig.IntPropertyFnWithNamespaceFilter
+		maxWorkflowTaskPollers       dynamicconfig.IntPropertyFnWithNamespaceFilter
+	}
+
+	activityDeps struct {
+		fx.In
+		MetricsClient metrics.Client
+		Logger        log.Logger
+		ClientFactory sdk.ClientFactory
+	}
+
+	fxResult struct {
+		fx.Out
+		Component workercommon.PerNSWorkerComponent `group:"perNamespaceWorkerComponent"`
+	}
+)
+
+var Module = fx.Options(
+	fx.Provide(NewResult),
+)
+
+func NewResult(
+	dc *dynamicconfig.Collection,
+	params activityDeps,
+) fxResult {
+	return fxResult{
+		Component: &workerComponent{
+			activityDeps:                 params,
+			dc:                           dc,
+			enabledFeature:               dc.GetBoolPropertyFnWithNamespaceFilter(dynamicconfig.EnableBatcher, true),
+			enabledPerNSWorker:           dc.GetBoolPropertyFnWithNamespaceFilter(dynamicconfig.EnableBatcherPerNamespaceWorker, false),
+			numWorkers:                   dc.GetIntPropertyFilteredByNamespace(dynamicconfig.WorkerBatcherPerNamespaceWorkerCount, 1),
+			maxActivityExecutionSize:     dc.GetIntPropertyFilteredByNamespace(dynamicconfig.WorkerBatcherMaxConcurrentActivityExecutionSize, 1000),
+			maxWorkflowTaskExecutionSize: dc.GetIntPropertyFilteredByNamespace(dynamicconfig.WorkerBatcherMaxConcurrentWorkflowTaskExecutionSize, 1000),
+			maxActivityPollers:           dc.GetIntPropertyFilteredByNamespace(dynamicconfig.WorkerBatcherMaxConcurrentActivityTaskPollers, 2),
+			maxWorkflowTaskPollers:       dc.GetIntPropertyFilteredByNamespace(dynamicconfig.WorkerBatcherMaxConcurrentWorkflowTaskPollers, 2),
+		},
+	}
+}
+
+func (s *workerComponent) DedicatedWorkerOptions(ns *namespace.Namespace) *workercommon.PerNSDedicatedWorkerOptions {
+	namespaceName := ns.Name().String()
+	enableFeature := s.enabledFeature(namespaceName)
+	enablePerNSWorker := s.enabledPerNSWorker(namespaceName) || ns.Name().String() == common.SystemLocalNamespace
+	return &workercommon.PerNSDedicatedWorkerOptions{
+		Enabled:    enableFeature && enablePerNSWorker,
+		TaskQueue:  BatcherTaskQueueName,
+		NumWorkers: s.numWorkers(namespaceName),
+		Options: sdkworker.Options{
+			MaxConcurrentActivityExecutionSize:     s.maxActivityExecutionSize(namespaceName),
+			MaxConcurrentWorkflowTaskExecutionSize: s.maxWorkflowTaskExecutionSize(namespaceName),
+			MaxConcurrentActivityTaskPollers:       s.maxActivityPollers(namespaceName),
+			MaxConcurrentWorkflowTaskPollers:       s.maxWorkflowTaskPollers(namespaceName),
```

Comment:
These aren't possible to set anymore since the worker is shared across all components. If necessary, you can set options in pernamespaceworker.go. I don't think we'll have too many so we can figure out values that make sense for all of them

Reply:
removed after merged master